src/app/shared/user-spaces.service.ts

```diff
@@ -0,0 +1,111 @@
+import {
+  HttpClient,
+  HttpErrorResponse,
+  HttpHeaders
+} from '@angular/common/http';
+import {
+  ErrorHandler,
+  Inject,
+  Injectable
+} from '@angular/core';
+import {
+  forkJoin,
+  Observable,
+  of,
+  Subscription
+} from 'rxjs';
+import {
+  catchError,
+  filter,
+  map,
+  switchMap,
+  zip
+} from 'rxjs/operators';
+
+import { Logger } from 'ngx-base';
+import { Context, Space, SpaceService, WIT_API_URL } from 'ngx-fabric8-wit';
+import { AuthenticationService } from 'ngx-login-client';
+
+export interface UserSpacesResponse {
+  data: SpaceInformation[];
+  meta: {
+    totalCount: number
+  };
+}
+
+export class SpaceInformation {
+  attributes: {
+    name: string
+  };
+  id: string;
+  links: {
+    self: string
+  };
+  type: string;
+}
+
+@Injectable()
+export class UserSpacesService {
+
+  private readonly headers: HttpHeaders = new HttpHeaders();
+
+  constructor(
+    private readonly http: HttpClient,
+    @Inject(WIT_API_URL) private readonly witUrl: string,
+    private readonly auth: AuthenticationService,
+    private readonly errorHandler: ErrorHandler,
+    private readonly logger: Logger,
+    private readonly spaceService: SpaceService
+  ) { }
+
+  getInvolvedSpacesCount(): Observable<number> {
+    let headers: HttpHeaders = this.headers;
+    if (this.auth.getToken() != null) {
+      headers = this.headers.set('Authorization', `Bearer ${this.auth.getToken()}`);
+    }
+    return this.http.get(`${this.witUrl}user/spaces`, { headers })
+      .pipe(
+        map((response: UserSpacesResponse) => response.meta.totalCount),
+        catchError((err: HttpErrorResponse): Observable<number> => {
+          this.errorHandler.handleError(err);
+          this.logger.error(err);
+          return of(0);
+        })
+      );
+  }
+
+    // Currently the backend returns values that look like Space[], but isn't
+    getInvolvedSpaces(): Observable<SpaceInformation[]> {
+      let headers: HttpHeaders = this.headers;
+      if (this.auth.getToken() != null) {
+        headers = this.headers.set('Authorization', `Bearer ${this.auth.getToken()}`);
+      }
+      return this.http.get(`${this.witUrl}user/spaces`, { headers })
+        .pipe(
+          map((response: UserSpacesResponse): SpaceInformation[] => response.data),
+          catchError((err: HttpErrorResponse): Observable<Space[]> => {
+            this.errorHandler.handleError(err);
+            this.logger.error(err);
```

Comment:
`getInvolvedSpaces` and `getInvolvedSpacesCount` can/should be refactored so the implementation is shared - they really only differ by the `map` applied in the pipe, and a slight difference in error handling return values that could easily be pulled out into an error handler function as well.